pages/404.tsx

```diff
@@ -0,0 +1,9 @@
+import type { NextPage } from 'next';
+
+const NotFound: NextPage = () => {
```

Comment:
Maybe just add a comment below here explaining we want to check if this is running on the server or not.

Reply:
This is running on the server in the [preview deployment URL](https://careers-git-feature-25-404-page-careers-site.vercel.app/). I'm not sure if this is an optimal solution as it loads a webpage before redirecting to the next webpage, I will see if we can avoid this.

Reply:
This flash looks to be unavoidable, if this was an SSR site their might have been more options avaliable to try and prevent this, i.e. `if (res.statusCode === 404) res.writeHead(404, {Location: 'https://torchbox.com/404/'})`

There are three points to mention though:

1) We can use this nextjs-redirect library for client redirects, if you look at it's code it seems to handle things more correctly https://github.com/pablopunk/nextjs-redirect/blob/master/src/index.tsx.

2) If you know of any paths we want to redirect outside of these that are already being handled on the current site (i.e. career urls under [wagtail redirects](https://docs.wagtail.io/en/stable/editor_manual/managing_redirects.html) on torchbox.com if there is any). then we can do them manually in the nextjs config: https://nextjs.org/docs/api-reference/next.config.js/redirects so we don't lose them when this goes live.

3) I thought we could update `getStaticProps` and add redirects there on other branches, but that only supports 300 status errors and I'm apprehensive about sending a 302 to a 404? otherwise it might be fine to as I guess we're still sending a 200 status before the redirect... 
```
    if (!job)
        return {
            // remove ---> notFound: true, 
            redirect: {
                 destination: 'https://torchbox.com/404/',
                 permanent: false
            }
        };
```

point 3 makes me think we should just have our own 404 page to simplify things, perhaps copying the one we're redirecting too?

Reply:
I've double checked point 2 about redirects with Dan, there aren't any that are relevant to the new /careers subdirectory. The `/culture-and-jobs` redirect leads to `/careers` already.